src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs

```diff
@@ -163,5 +167,20 @@ public async Task<ActionResult> OnPostToggleClosedAsync(string id)
 
             return RedirectToPage(new { id = id });
         }
+
+        public async Task<ActionResult> OnPostToggleSubscribedAsync(string id)
+        {
+            ReviewModel review = await _manager.GetReviewAsync(User, id);
+            if (review.IsUserSubscribed(User))
+            {
+                review.Unsubscribe(User);
+            }
+            else
+            {
+                review.Subscribe(User);
+            }
+            await _reviewRepository.UpsertReviewAsync(review);
```

Comment:
Don't use review repository directly, do it via the manager so all the security checks are applied.

Reply:
Calling manager.UpdateReviewAsync caused a Blob already exists error, I think do to it uploading the codefile again.

Reply:
UpdateReviewAsync is a bad name, what it does is re-generating code model after generator changes. I would put subscription all toggling logic into NotificationManager.

Reply:
Adding Upsert wrapper to NotificationManager. As discussed, leaving the toggling logic in ReviewModel.